CS2113-T13-3 Classify

ParthGandhiNUS opened the pull request.

Mmaxx15 commented on docs/DeveloperGuide.md:
```diff
@@ -2,21 +2,198 @@
+
+#### Implementation and rationale
+
+![InputParsingSequenceDigram](./diagrams/src/InputParsing/InputParsingSequenceDiagram.png)
```
is there suppose to be a X at the end of lifeline?

NicholasTanYY replied:
Good try on the sequence diagram! However, should the studentInfo.txt be omitted from the sequence diagram? It would be better if there is a method call to the class which handles the modification of studentInfo.txt instead.

NicholasTanYY commented on docs/DeveloperGuide.md:
```diff
@@ -2,21 +2,198 @@
+
+This component ensures that the user parses in commands in a format that makes sense, which will modify the master list.
+
+![InputParsingUML](./diagrams/src/InputParsing/InputParsing.png)
```
Should the green 'C' be removed from the UML diagram? It seems that the official convention does not use the green 'C'.

damiwee replied:
Yes, you can refer to the plant UML documentation to remove the circles.

Mmaxx15 commented on docs/DeveloperGuide.md:
```diff
@@ -2,21 +2,198 @@
+
+Seen below is an UML diagram of the relevant classes dealing with storage.
+
+![DataStoringUML](./diagrams/src/DataStoring/DateStoring.png)
```
Good job in adding the UML diagram to help with visualisation

NicholasTanYY replied:
Same issue here, would removing the green 'C' be better?

Mmaxx15 commented on docs/DeveloperGuide.md:
```diff
@@ -2,21 +2,198 @@
-{Describe the value proposition: what problem does it solve?}
+Classify serves as an attempt to modernise administrative tasks in education institutes, such as tuition centres or school environments.
 
 ## User Stories
```
Could have more user stories which have more specific functionality requirements, current one seems quite vague
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/88656341/920de52c-8ac0-414c-b201-a0502c659bbc)

Mmaxx15 commented on docs/DeveloperGuide.md:
```diff
@@ -35,4 +212,21 @@
 
```
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/88656341/d6bc139b-72ec-4877-8eec-4301ce1d4c61)
Nun functional requirements is empty, can specify the constraints under which the system is developed and operated

Mmaxx15 commented on docs/DeveloperGuide.md:
```diff
@@ -35,4 +212,21 @@
 ## Instructions for manual testing
 
-{Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing}
+### Adding a student to the student list
```
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/88656341/2fd2fef9-4662-43a7-9e52-5d0342e7a2f2)
This format seems abit cluttered, could use screenshots to help with visualisation

NicholasTanYY commented on docs/DeveloperGuide.md:
```diff
@@ -2,21 +2,198 @@
+
+Please see the diagram below to see how identifying a student works.
+
+![studentIdentification](./diagrams/src/DataStoring/Student%20Identification.png)
```
Great attempt! I think some of the method calls to various classes seems to be returning multiple objects in the same activation bar, when there is only the initial input method call. Perhaps what you intend to show is 2 activation bars, each returning one object. Or within the same activation bar, there should be some input before there is a return value?

Also should the arrows which show the returning from the activation bar be dotted?

NicholasTanYY commented on docs/DeveloperGuide.md:
```diff
@@ -35,4 +212,21 @@
+   Expected: `view joe` now shows the Student details of a student with Name: joe, Phone Number: 11111111. Other fields that were left blank will reflect 'Unknown' or for date fields, today's date.
+   3. Test case: `add` and when prompted for Name, `joe`. `11111111` when prompted for phone number, press enter to skip other fields.
+   <br />
+   Expected: `view joe` shows the same results as when a student was added via `add joe`.
```
Adding a <br> tag here might help with the indentation issue!

NicholasTanYY commented on docs/DeveloperGuide.md:
```diff
@@ -35,4 +212,21 @@
+   3. Test case: `add` and when prompted for Name, `joe`. `11111111` when prompted for phone number, press enter to skip other fields.
+   <br />
+   Expected: `view joe` shows the same results as when a student was added via `add joe`.
+### Viewing a student's details
+1. View a student who has been added to the student list
+   1. Prerequisites: Add one student named 'joe' to the list with the `add` command
+   2. Test case: `view joe`
+   <br />
+   Expected: Student's details shown correspond to the details input when `add` was used to add a student.
+   3. Test case: `view dogman`
+   <br />
+   Expected: No details are displayed, an error message stating 'Student not found!' is shown.
```
Perhaps this section can be supported with code blocks showing the expected outputs that corresponds to the inputs given instead to make things clearer?

NicholasTanYY reviewed (commented):
Overall, the developer guide looks good! It is detailed and sufficient for a developer to understand the main structure of the code. However, please note some of the UML diagram improvements that could be considered. LGTM

PureUsagi commented on docs/DeveloperGuide.md:
```diff
@@ -2,28 +2,206 @@
+
+- [Source](https://se-education.org/addressbook-level3/DeveloperGuide.html#proposed-undoredo-feature)
+- Used as template to structure this Developer Guide
 
```
Shouldn't there be a table of contents to give readers easy access to various parts of the developer guide? Maybe putting one can help with the navigability of the guide!

damiwee replied:
I agree, a table of contents should be added.

PureUsagi commented on docs/DeveloperGuide.md:
```diff
@@ -2,28 +2,206 @@
+
+#### Implementation and rationale
+
+![InputParsingSequenceDigram](./diagrams/src/InputParsing/InputParsingSequenceDiagram.png)
```
I think the stickman and database image are unnecessary for the sequence diagram. Actually, shouldn't the database image be omitted as its irrelevant to the course?

damiwee replied:
User does not need a ":"

HiteshriAcharya replied:
I think the activation bar is incomplete

HiteshriAcharya replied:
Please follow the format of the sequence diagram hehe and maybe the images in the middle might not be needed?

PureUsagi commented on docs/DeveloperGuide.md:
```diff
@@ -35,4 +213,21 @@
 
 ## Instructions for manual testing
```
Maybe the manual testing section can have more commands to be more comprehensive (e.g delete, list, or edit)? Rather than testing each command in isolation, maybe can group them together for a single test(e.g show add first, then list, then delete, then list, etc......)?

PureUsagi commented on docs/DeveloperGuide.md:
```diff
@@ -2,28 +2,206 @@
+
+Please see the diagram below to see how identifying a student works.
+
+![studentIdentification](./diagrams/src/DataStoring/Student%20Identification.png)
```
Shouldn't the entity names be strictly classes? Having the names as methods like findStudentByName and findStudentByNumber might not follow the sequence diagram standard.

PureUsagi reviewed (commented):
Overall great work on the develop guide, but there may some errors with the sequence diagrams.

PureUsagi commented on docs/DeveloperGuide.md:
```diff
@@ -2,28 +2,206 @@
+The `StudentSorter` class facilitates sorting of the list of students based on various criterion.
+As of the latest iteration, it sorts the master list of students in the order specified by the user.
+
+![StudentSorterUML](./diagrams/src/StudentSorter/StudentSorter.png) 
```
Similar to other diagrams in the develop guide, shouldn't the PlantUML notations / formatting be disabled (e.g. access modifier, green 'C')?

PureUsagi reviewed (commented):
One more extra comment on one of the diagrams

damiwee commented on docs/DeveloperGuide.md:
```diff
@@ -2,28 +2,206 @@
+
+![InputParsingUML](./diagrams/src/InputParsing/InputParsing.png)
+
+#### Design considerations
+- The parser must be effecive in breaking down the user's input into its constitutent commands, with further breakdown if an associated argument is added.
+- The parser should be quick and effective in understanding the user's input, with simple prompts given to help the user in correctly parsing the command through the input parser.
+- Allowing the user to input optional arguments. For example, the user could type `view <student name>`, which takes in the "student name" as an optional argument. This is to increase the robustness of the program, which accounts for the two types of users, one who just types in `view`, and the other as formerly mentioned above. 
+- Error handling must be intuitive for the user, so that appropriate error messages are produced if the user does not input a valid command. The error handling should also be robust, to account for the event a user is incapable in following instructions.
```
Perhaps consider splitting into user and developer design considerations

damiwee commented on docs/DeveloperGuide.md:
```diff
@@ -2,28 +2,206 @@
-{Describe the value proposition: what problem does it solve?}
+Classify serves as an attempt to modernise administrative tasks in education institutes, such as tuition centres or school environments.
 
 ## User Stories
```
User stories seems incomplete

damiwee reviewed (commented):
Overall, seems like you guys are on the right path, just need to fill in the missing components and some tiny errors in the diagrams.

HiteshriAcharya commented on docs/DeveloperGuide.md:
```diff
@@ -2,28 +2,206 @@
+
+This component ensures that the user parses in commands in a format that makes sense, which will modify the master list.
+
+![InputParsingUML](./diagrams/src/InputParsing/InputParsing.png)
```
Please disable the symbols "C"

HiteshriAcharya replied:
Wow, you are using a package diagram! Maybe, the use of a Package Diagram might be confusing?

blackmirag3 committed:
Update UG

blackmirag3 committed:
Fix formatting for UG

blackmirag3 committed:
Update expected test case output

ParthGandhiNUS committed:
Make private class public for usage in other parts of codebaase

ParthGandhiNUS committed:
Update process command workflow to reduce double counting in MasterStudentList and SubjectLists

ParthGandhiNUS committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-T13-3/tp into ParseTextFile

alalal47 committed:
Merge pull request #266 from ParthGandhiNUS/ParseTextFile

Update functionality for the process command

ParthGandhiNUS committed:
Fix bug for the max number of classes attended

ParthGandhiNUS committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-T13-3/tp into Bug_Fixing

ParthGandhiNUS committed:
Fix small bug

alalal47 committed:
Merge pull request #267 from ParthGandhiNUS/Bug_Fixing

Bug fixing

alalal47 committed:
Add handling for cases of file corruption that are not handled by our other methods.
Add option for user to delete file in cases where they are unable to manually rectify corruption by editing the info file(that they messed up)

alalal47 committed:
Fix wrong indentation for comment

alalal47 committed:
Fix incorrect indentation

tayponghee committed:
Merge pull request #268 from alalal47/corruption_handling

Add handling for cases of file corruption that are not handled by our…

tayponghee committed:
Update README file to match our product

tayponghee committed:
Update PPP

tayponghee committed:
Update Junit tests

tayponghee committed:
Delete unused package

tayponghee committed:
Remove unneccessary test class, update inputparsingtest

tayponghee committed:
Update StudentAttributesTest

tayponghee committed:
Update PPP

tayponghee committed:
Remove extraneous bracket

ParthGandhiNUS committed:
Merge pull request #271 from tayponghee/update_aboutus

Update README file to match our product

tayponghee committed:
Add classes attended for the specific subject in the list command

tayponghee committed:
Update UG

tayponghee committed:
Fix gradle checks and Junit tests

alalal47 committed:
Fix file always corrupted bug

alalal47 committed:
Fix incorrect indentation

tayponghee committed:
Merge pull request #273 from alalal47/junit

Fix file always corrupted bug

alalal47 committed:
Merge pull request #272 from tayponghee/update_aboutus

Add classes attended for each subject

alalal47 committed:
Implement clearing of student list upon corruption detected

tayponghee committed:
Merge pull request #274 from alalal47/junit

Implement clearing of student list upon corruption detected

alalal47 committed:
Add testing for sort by payment date

alalal47 committed:
Fix line wrapping for help message on smaller terminal windows

alalal47 committed:
Remove unused import

blackmirag3 committed:
Update UserGuide.md

blackmirag3 committed:
Merge branch 'master' into shui-edit-v2

alalal47 committed:
Merge pull request #265 from blackmirag3/shui-edit-v2

Post-mock PE UG updates

tayponghee committed:
Merge pull request #275 from alalal47/junit

Junit tests and edit help function to prevent line wrapping

ParthGandhiNUS committed:
Update TextFileHandler Class

ParthGandhiNUS committed:
Update Main of Classify

ParthGandhiNUS committed:
Update access of TextFileParser methods

ParthGandhiNUS committed:
Junit tests for TextFileHandler

ParthGandhiNUS committed:
Junit tests for TextFileReader

ParthGandhiNUS committed:
Junit tests for TextFileParser

ParthGandhiNUS committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-T13-3/tp into V2.1

ParthGandhiNUS committed:
Correct CheckStyle Issues

ParthGandhiNUS committed:
Remove erronous test case

alalal47 committed:
Merge pull request #276 from ParthGandhiNUS/V2.1

Add Junit Test

alalal47 committed:
Add more features to instructions for manual testing

tayponghee committed:
Remove unused imports

alalal47 committed:
Fix instructions for manual testing formatting

Cryolian committed:
Change StudentSorter image

ParthGandhiNUS committed:
Merge pull request #279 from Cryolian/master

Change StudentSorter image

Cryolian committed:
Merge pull request #278 from tayponghee/master

Remove unused imports

alalal47 committed:
Fix newlines and add restore instructions for manual testing

alalal47 committed:
Add undo instructions for manual testing

Cryolian committed:
Change UG intro

ParthGandhiNUS committed:
Merge pull request #280 from Cryolian/bugfixv2.0

Change UG intro

Cryolian committed:
Change classes attended message

blackmirag3 committed:
Add edit manual testing cases for DG

Cryolian committed:
Fix file writing bug

Cryolian committed:
Add phone number checks to save files.

Cryolian committed:
Fix user guide (process)

Cryolian committed:
Fix sort infinite loop

Cryolian committed:
Fix tests and printing messages

Cryolian committed:
Fix parth's poopoo brackets

Cryolian committed:
Clean parth's peepee

Cryolian committed:
Fix checkstyle errors

Cryolian committed:
Fix spacing

tayponghee committed:
Change name

ParthGandhiNUS committed:
Merge pull request #282 from Cryolian/bugfixv2.0

Im down with a cold

Cryolian committed:
Merge pull request #281 from blackmirag3/shui-edit-v2

Add edit manual testing cases for DG

tayponghee committed:
Refactor addStudent

tayponghee committed:
Fix author tag

ParthGandhiNUS committed:
Make classes for DataHandler, DataReader, and DataWriter

ParthGandhiNUS committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-T13-3/tp into V2.1

tayponghee committed:
Refactor UI message

tayponghee committed:
Change boolean nomenclature

tayponghee committed:
Remove unused constant

tayponghee committed:
Change all constants to include _ for "yi xin"

tayponghee committed:
Change xiao_ming

ParthGandhiNUS committed:
Merge pull request #283 from tayponghee/master

Change name

ParthGandhiNUS committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-T13-3/tp into V2.1

alalal47 committed:
Update DG

alalal47 committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-T13-3/tp into DG

# Conflicts:
#	docs/DeveloperGuide.md
#	docs/UserGuide.md

alalal47 committed:
Update DG

alalal47 committed:
Fix spelling error

alalal47 committed:
Remove extraneous symbols

tayponghee committed:
Merge pull request #277 from alalal47/DG

Add more features to instructions for manual testing

blackmirag3 committed:
Update blackmirag3.md

blackmirag3 committed:
Update blackmirag3.md

ParthGandhiNUS committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-T13-3/tp into V2.1

ParthGandhiNUS committed:
Small Change in Naming Test Method

ParthGandhiNUS committed:
Change Access Modifiers for methods

ParthGandhiNUS committed:
Add DataReader Test cases

ParthGandhiNUS committed:
Add Datawriter Test cases

ParthGandhiNUS committed:
Change the numbering for test case

tayponghee committed:
Merge pull request #285 from ParthGandhiNUS/V2.1

Add Test Cases for code for Data

Cryolian committed:
Add view missing student test

Cryolian committed:
Add number and gender tests

Cryolian committed:
Fix checkstyle errors

Cryolian committed:
Fix test params

Cryolian committed:
Change test checks

Cryolian committed:
Merge pull request #286 from Cryolian/bugfixv2.0

Add unit tests for editing gender and number

Cryolian committed:
Add archive tests and Fix process bugs

Cryolian committed:
Adjust archive tests

Cryolian committed:
Fix archive tests

alalal47 committed:
Merge pull request #287 from Cryolian/bugfixv2.0

Add archive tests and Fix process bugs

blackmirag3 committed:
Update blackmirag3.md

blackmirag3 committed:
Merge branch 'shui-edit-v2' of github.com:blackmirag3/tp into shui-edit-v2

tayponghee committed:
Add architectural diagram, update DG

tayponghee committed:
Update PPP

blackmirag3 committed:
Update DG

Cryolian committed:
Merge pull request #288 from tayponghee/Architecture_diagram

Add architectural diagram, update DG

Cryolian committed:
Add portfolio links to aboutus

Cryolian committed:
Update UG to make date storage details explicit

Cryolian committed:
Update aboutus links

blackmirag3 committed:
Update blackmirag3.md

tayponghee committed:
Merge pull request #290 from Cryolian/bugfixv2.0

Bugfixv2.0

tayponghee committed:
Change AboutUs, modified link to not show on the page

Cryolian committed:
Merge pull request #291 from tayponghee/master

Change AboutUs, modified link to not show on the page

ParthGandhiNUS committed:
Update PPP

alalal47 committed:
Merge pull request #292 from ParthGandhiNUS/V2.1

Update PPP

ParthGandhiNUS committed:
Add table of content to DG

ParthGandhiNUS committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-T13-3/tp into V2.1

alalal47 committed:
Merge pull request #293 from ParthGandhiNUS/V2.1

Update DG

alalal47 committed:
Clean up code and update help function to be more readable on small screens

alalal47 committed:
Fix naming for magic literal

alalal47 committed:
Fix examples in ug to match product

Cryolian committed:
Merge pull request #294 from alalal47/DG

Clean up code and update help function to be more readable on small s…

tayponghee committed:
Remove circle from Architecture Diagram

alalal47 committed:
Merge pull request #295 from tayponghee/master

Remove circle from Architecture Diagram

alalal47 committed:
Add support for small window/screen/big font size to the help feature

alalal47 committed:
Reformat code to make it more readable

alalal47 committed:
Update alalal47.md and UG to reflect latest version of help function

alalal47 committed:
Fix typos and junit

Cryolian committed:
Merge pull request #296 from alalal47/help_revamp

Help revamp

ParthGandhiNUS committed:
Edit Test Case to remove empty directory which is created

ParthGandhiNUS committed:
Add more test cases for TextFileParser

alalal47 committed:
Update UG with fixed typos in help function

ParthGandhiNUS committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-T13-3/tp into V2.1

alalal47 committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-T13-3/tp into help_revamp

alalal47 committed:
Merge pull request #297 from alalal47/help_revamp

Help revamp

ParthGandhiNUS committed:
Update DG

ParthGandhiNUS committed:
Update UG

ParthGandhiNUS committed:
Add SampleFile for Process Command

ParthGandhiNUS committed:
Add Image for the format of the text file

ParthGandhiNUS committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-T13-3/tp into V2.1

ParthGandhiNUS committed:
Update code to meet checkstyle requirements

tayponghee committed:
Merge pull request #298 from ParthGandhiNUS/V2.1

Update UG, DG, Test Cases

blackmirag3 committed:
Merge branch 'master' into shui-edit-v2

ParthGandhiNUS committed:
Update image for About Us Page

ParthGandhiNUS committed:
Update PPP

ParthGandhiNUS committed:
Update UG

ParthGandhiNUS committed:
Clean everyone's peepee poopoo for boolean stuff

ParthGandhiNUS committed:
Update DG for Process Component

Cryolian committed:
Merge pull request #299 from ParthGandhiNUS/V2.1

Update DG, PPP, About Us. Clean Code

Cryolian committed:
Merge pull request #284 from blackmirag3/shui-edit-v2

Update blackmirag3.md

blackmirag3 committed:
Add logging to archive commands

ParthGandhiNUS committed:
Remove redundant println

ParthGandhiNUS committed:
Update DG with manual testing for Process

ParthGandhiNUS committed:
Refactor process functionality to handle bugs due to different filetypes

ParthGandhiNUS committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-T13-3/tp into V2.1

blackmirag3 committed:
Add exceptions for editing subject

Cryolian committed:
Merge pull request #300 from ParthGandhiNUS/V2.1

Update DG, Bug Fix Process Functionality

tayponghee committed:
Update UG and sort function UI

ParthGandhiNUS committed:
Merge pull request #301 from tayponghee/master

Update UG and sort function UI

tayponghee committed:
Update UG

Cryolian committed:
Merge pull request #302 from tayponghee/master

Update UG

Cryolian committed:
Test photo fix

Cryolian committed:
Fix image size

blackmirag3 committed:
Update PPP

Cryolian committed:
Add image

ParthGandhiNUS committed:
Edit PPP

blackmirag3 committed:
Merge branch 'master' into shui-edit-v2

blackmirag3 committed:
Update AddStudent.java

Cryolian committed:
Merge pull request #304 from blackmirag3/shui-edit-v2

Update DG

Cryolian committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-T13-3/tp into bugfixv2.0

Cryolian committed:
Fix spacing

Cryolian committed:
Fix spacing

Cryolian committed:
Another fix attempt

alalal47 committed:
Add photo to aboutus.md

alalal47 committed:
Merge pull request #305 from alalal47/help_revamp

Add photo to aboutus.md

Cryolian committed:
Merge branch 'master' into bugfixv2.0

Cryolian committed:
Merge pull request #306 from Cryolian/bugfixv2.0

Image fix attempt

ParthGandhiNUS committed:
Add pagebreaks for UG

ParthGandhiNUS committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-T13-3/tp

Cryolian committed:
Another fix attempt

Cryolian committed:
Merge branch 'master' into bugfixv2.0

Cryolian committed:
Another fix attempt

ParthGandhiNUS committed:
Merge pull request #307 from ParthGandhiNUS/master

Update Docu

Cryolian committed:
Remove space

Cryolian committed:
Add lines

Cryolian committed:
Add missing line

Cryolian committed:
insn

Cryolian committed:
last try before i do html

Cryolian committed:
1

Cryolian committed:
12

ParthGandhiNUS committed:
Update UG to make it more printable

ParthGandhiNUS committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-T13-3/tp

ParthGandhiNUS committed:
Merge pull request #308 from ParthGandhiNUS/master

Update UG

Cryolian committed:
html attempt

Cryolian committed:
Merge pull request #309 from Cryolian/bugfixv2.0

Fix about us page

ParthGandhiNUS committed:
Add a few more test cases to be more comprehensive

ParthGandhiNUS committed:
Merge pull request #310 from ParthGandhiNUS/V2.1

Add a few more test cases to be more comprehensive

tayponghee committed:
Add exception handling and remove unrelated NFR

tayponghee committed:
Add nullpointerexception handling, fix author tag

tayponghee committed:
Update UG

tayponghee committed:
Fix errorneous add example location

ParthGandhiNUS committed:
Merge pull request #311 from tayponghee/master

Update UG, add exception handling

tayponghee committed:
Update UG

ParthGandhiNUS committed:
Merge pull request #312 from tayponghee/master

Update UG

blackmirag3 committed:
Add archive component to Dg

blackmirag3 committed:
Update UG and PPP

blackmirag3 committed:
Update typos in DG

ParthGandhiNUS committed:
Edit UG

blackmirag3 committed:
Update DG

blackmirag3 committed:
Fix typo in DG

alalal47 committed:
Merge pull request #313 from blackmirag3/shui-edit-v2

Add archive component to Dg

alalal47 committed:
Fix typos in DG

Cryolian committed:
Merge pull request #314 from alalal47/help_revamp

Fix typos in DG

ParthGandhiNUS committed:
UG is frustrating

ParthGandhiNUS committed:
UG is not stonks

ParthGandhiNUS committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-T13-3/tp into V2.1

Cryolian committed:
Merge pull request #315 from ParthGandhiNUS/V2.1

UG Finale

ParthGandhiNUS committed:
Update PPP

alalal47 committed:
Merge pull request #316 from ParthGandhiNUS/V2.1

Update PPP

Cryolian committed:
Update DeveloperGuide.md

Cryolian committed:
Update DeveloperGuide.md

Cryolian committed:
Merge pull request #317 from AY2324S2-CS2113-T13-3/Cryolian-patch-2

Update DeveloperGuide.md

ParthGandhiNUS committed:
Update DG

ParthGandhiNUS committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-T13-3/tp into V2.1

# Conflicts:
#	docs/DeveloperGuide.md

ParthGandhiNUS committed:
Update DG

ParthGandhiNUS committed:
Update DG

ParthGandhiNUS committed:
Update DG

ParthGandhiNUS committed:
Update DG

tayponghee committed:
Merge pull request #318 from ParthGandhiNUS/master

Update DG

Cryolian committed:
test fix

ParthGandhiNUS committed:
Add User Story

Cryolian committed:
Merge pull request #319 from ParthGandhiNUS/master

Add User Story

Cryolian committed:
fix apos

Cryolian committed:
typo

Cryolian committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-T13-3/tp into bugfixv2.0

Cryolian committed:
change back to 6

Cryolian committed:
change enter

Cryolian committed:
change back

Cryolian committed:
Merge pull request #320 from Cryolian/bugfixv2.0

Bugfixv2.0

Cryolian committed:
Update DeveloperGuide.md
